core/nodefactory.lua

```diff
@@ -306,8 +306,9 @@ local _vbox = _box {
     typesetter.frame:advancePageDirection(advanceamount)
     local initial = true
     for _, node in pairs(self.nodes) do
+      -- luacheck: ignore 542
+      -- (It's easier to understand a do-nothing branch than the logical inverse)
       if initial and (node:isGlue() or node:isPenalty()) then
```

Comment:
Is it? I definitely agree with the concept that code which is easier to read and understand is better even if there are technically better language language semantics....but I tend to think this is actually _more_ readable, dropping the `else` and just using the `if` block naturally:

```suggestion
      if initial and not node:isGlue() and not node:isPenalty()) then
```

Note the value handling of initial might need editing elsewhere for this flow. I'm not saying it has to go into this PR, just that in general the logic of "if A and (B or C) then nil else ..." is actually pretty confusing to me.